inc/saibfd.h

```diff
@@ -0,0 +1,563 @@
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saibfd.h
+ *
+ * @brief   This module defines SAI BFD interface
+ */
+
+#if !defined (__SAIBFD_H_)
+#define __SAIBFD_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIBFD SAI - BFD specific public APIs and data structures
+ *
+ * @{
+ */
+
+/**
+ * @brief SAI session type of BFD
+ */
+typedef enum _sai_bfd_session_type_t
+{
+    /** Demand Active Mode */
+    SAI_BFD_SESSION_TYPE_DEMAND_ACTIVE = 0,
+
+    /** Demand Passive Mode */
+    SAI_BFD_SESSION_TYPE_DEMAND_PASSIVE,
+
+    /** Asynchronous Active Mode */
+    SAI_BFD_SESSION_TYPE_ASYNC_ACTIVE,
+
+    /** Asynchronous Passive Mode */
+    SAI_BFD_SESSION_TYPE_ASYNC_PASSIVE,
+
+} sai_bfd_session_type_t;
+
+/**
+ * @brief SAI type of encapsulation for BFD
+ */
+typedef enum _sai_bfd_encapsulation_type_t
+{
+    /**
+     * @brief IP in IP Encapsulation | L2 Ethernet header | IP header | Inner IP header | Original BFD packet
+     */
+    SAI_BFD_ENCAPSULATION_TYPE_IP_IN_IP,
+
+    /**
+     * @brief L3 GRE Tunnel Encapsulation | L2 Ethernet header | IP header | GRE header | Original BFD packet
+     */
+    SAI_BFD_ENCAPSULATION_TYPE_L3_GRE_TUNNEL,
+
+} sai_bfd_encapsulation_type_t;
+
+/**
+ * @brief SAI BFD session state
+ */
+typedef enum _sai_bfd_session_state_t
+{
+    /** BFD Session is in Admin down */
+    SAI_BFD_SESSION_STATE_ADMIN_DOWN,
+
+    /** BFD Session is Down */
+    SAI_BFD_SESSION_STATE_DOWN,
+
+    /** BFD Session is in Initialization */
+    SAI_BFD_SESSION_STATE_INIT,
+
+    /** BFD Session is Up */
+    SAI_BFD_SESSION_STATE_UP,
+
+} sai_bfd_session_state_t;
+
+/**
+ * @brief Defines the operational status of the BFD session
+ */
+typedef struct _sai_bfd_session_state_notification_t
+{
+    /** BFD Session id */
+    sai_object_id_t bfd_session_id;
+
+    /** BFD session state */
+    sai_bfd_session_state_t session_state;
+
+} sai_bfd_session_state_notification_t;
+
+/**
+ * @brief SAI attributes for BFD session
+ */
+typedef enum _sai_bfd_session_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_BFD_SESSION_ATTR_START,
+
+    /**
+     * @brief BFD Session type DEMAND/ASYNCHRONOUS
+     *
+     * @type sai_bfd_session_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_BFD_SESSION_ATTR_TYPE = SAI_BFD_SESSION_ATTR_START,
+
+    /**
+     * @brief Hardware lookup valid
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
```

Comment:
can you make the default to be true since that is more desired mode?

Reply:
Changed